### ISSUE TITLE
Conflicting overloads, redeclarations, overload resolution ambiguity since version 1.5.0

### ISSUE THREAD

**Issue by HylkeB** [Kodein-Framework/MocKMP#23]
When running the gradle command `./gradlew testReleaseUnitTest testDebugUnitTest` (or running multiple tests of different modules in 1 command), I get compile issues (Redeclarations, overload resolution ambiguity)

My test class is just a very simple basic UnitTest with mocks, and I am trying to execute the tests of multiple flavors/configurations at once. Running this at versions 1.1.0 to 1.4.0 of MocKMP gives no issues. When using version 1.5.0 or 1.6.0 i get the issues.

In the build folders i see duplicate generated code:
```
build
  \ generated
    \ ksp
      \ android
        \ androidDebugUnitTest
          \ <kotlin/package structure and generated code. IDE says the kotlin folder is part of [commonTest]>
        \ androidReleaseUnitTest
          \ <kotlin/package structure and generated code. IDE says the kotlin folder is part of [commonTest]>
```

These are the issues i am seeing

```

> Task :compileDebugUnitTestKotlinAndroid FAILED
Execution optimizations have been disabled for task ':compileDebugUnitTestKotlinAndroid' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: '/<my_project_location>/build/generated/ksp/android/androidReleaseUnitTest/kotlin'. Reason: Task ':compileDebugUnitTestKotlinAndroid' uses this output of task ':kspReleaseUnitTestKotlinAndroid' without declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.4.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
e: /<my_project_location>/build/generated/ksp/android/androidDebugUnitTest/kotlin/com/example/HelloCoreTest_injectMocks.kt: (6, 1): Conflicting overloads: internal fun HelloCoreTest.injectMocks(mocker: Mocker): Unit defined in com.example in file HelloCoreTest_injectMocks.kt, internal fun HelloCoreTest.injectMocks(mocker: Mocker): Unit defined in com.example in file HelloCoreTest_injectMocks.kt
e: /<my_project_location>/build/generated/ksp/android/androidDebugUnitTest/kotlin/com/example/MockHelloBrowser.kt: (6, 16): Redeclaration: MockHelloBrowser
e: /<my_project_location>/build/generated/ksp/android/androidDebugUnitTest/kotlin/com/example/MockHelloNetworking.kt: (6, 16): Redeclaration: MockHelloNetworking
e: /<my_project_location>/build/generated/ksp/android/androidReleaseUnitTest/kotlin/com/example/HelloCoreTest_injectMocks.kt: (6, 1): Conflicting overloads: internal fun HelloCoreTest.injectMocks(mocker: Mocker): Unit defined in com.example in file HelloCoreTest_injectMocks.kt, internal fun HelloCoreTest.injectMocks(mocker: Mocker): Unit defined in com.example in file HelloCoreTest_injectMocks.kt
e: /<my_project_location>/build/generated/ksp/android/androidReleaseUnitTest/kotlin/com/example/MockHelloBrowser.kt: (6, 16): Redeclaration: MockHelloBrowser
e: /<my_project_location>/build/generated/ksp/android/androidReleaseUnitTest/kotlin/com/example/MockHelloNetworking.kt: (6, 16): Redeclaration: MockHelloNetworking
e: /<my_project_location>/src/commonTest/kotlin/com/example/HelloCoreTest.kt: (9, 33): Overload resolution ambiguity: 
internal fun HelloCoreTest.injectMocks(mocker: Mocker): Unit defined in com.example in file HelloCoreTest_injectMocks.kt
internal fun HelloCoreTest.injectMocks(mocker: Mocker): Unit defined in com.example in file HelloCoreTest_injectMocks.kt
```

**Comment by SalomonBrys** [comment]
Do you have a reproducer project ?

**Comment by HylkeB** [comment]
[multiplatform-project.zip](https://github.com/Kodein-Framework/MocKMP/files/8755406/multiplatform-project.zip)
I extracted the issue in this reproducer project.

In the `build.gradle.kts` file of the root project on line 20, you can change the version of the MocKMP dependency. version 1.4.0 works, but 1.5.0 and 1.6.0 do not.

To see that it malfunctions either run `./gradlew runUnitTests` (thats a convenience task that adds multiple unit test tasks together) or run them all in 1 command like so: `./gradlew :core:testReleaseUnitTest :core:iosX64Test :lib1:testReleaseUnitTest :lib1:iosX64Test :lib2:testReleaseUnitTest :lib2:iosX64Test`

**Comment by SalomonBrys** [comment]
Fixed in 1.7.0

**Comment by HylkeB** [comment]
It does not work with my reproducer project in version 1.7.0 (when running `./gradlew runUnitTests`)
